tools/profiling/dbpreader.c

```diff
@@ -327,18 +329,24 @@ const dbp_event_t *dbp_iterator_current(dbp_event_iterator_t *it)
     return &it->current_event;
 }
 
-static const dbp_event_t *dbp_iterator_set_offset(dbp_event_iterator_t *it, off_t offset)
+static const dbp_event_t *dbp_iterator_set_offset(dbp_event_iterator_t *it,
```

Comment:
why do we need 3 variables to point us to a position in a stream ?

Reply:
1) which buffer
2) position inside buffer
3) index of this position (this isn't really needed, but can't be calculated)

Reply:
I was referring to `offset`, `event_pos` and `event_idx`. How do I come up with a tuple `offset, event_pos, event_idx` that makes sense, and how is the function reacting when I provide incorrect values ?

Reply:
I don't think I can (easily) sanity-check `offset`, unless the offset of the last valid buffer in the file is stored somewhere in the header.

Reply:
I have no way of checking whether the `event_pos` and `event_idx` agree with _each-other_, but that's the entire reason I need to cache both.

Reply:
ok, so in addition to moving to a new profiling page (using offset), you also move to a specific event (event_pos in bytes and event_idx in logical position). In fact, one could have used a single offset (using modulo to compute the relative displacement from the beginning of the current page).

Moreover, a short description would have saved time to read the code. So, this is all perfectly fine, for as long as the function does some checks and it is not called `dbp_iterator_set_offset`, but uses a more clear name, something that indicates that it moves to a specific event.

Reply:
> In fact, one could have used a single offset (using modulo to compute the relative displacement from the beginning of the current page).

I think I'm a bit unclear on what you mean by this—events can have different sizes due to info, so you can't easily associate event position and index without iterating over the prior events in the buffer.

Reply:
There are two offsets in use here: `offset` which is relative to the file and increases in increments of profiling page size, and `event_pos` which is relative to the starting of the current page and which is increment using the real size of the events that are read. These two offset are complementary and because the page size is of fixed size, and the other is always smaller than the page size, can be combined together.

Reply:
Oh I see what you mean, clever! I was thinking that since `event_pos` is relative not to the start of the page but to the event buffer (i.e. after the header) that it could cause issues, but you're right, it wouldn't. I'll implement that.